src/implementing_new_feature.md

```diff
@@ -0,0 +1,135 @@
+
+For other features, you'll have to make a tracking issue for that feature. The issue title should be "Tracking issue for YOUR FEATURE" and it should have the `B-unstable` & `C-tracking-issue` tags, along with the tag for your subteam (e.g. `T-compiler` if this is a compiler feature).
+
+For tracking issues for features (as opposed to future-compat warnings), I don't think the description has to contain anything specific. Generally we put the list of items required for stabilization using a github list, e.g.
+
+```bash, ignore
+    **Steps:**
+
+    - [ ] Implement the RFC (cc @rust-lang/compiler -- can anyone write up mentoring instructions?)
+    - [ ] Adjust documentation ([see instructions on forge][doc-guide])
+    - Note: no stabilization step here.
+```
+
+## Stability in code
+
+The below steps needs to be followed in order to implement a new unstable feature:
+
+1. Open a [tracking issue] - if you have an RFC, you can use the tracking issue for the RFC.
+
+2. Pick a name for the feature gate (for RFCs, use the name in the RFC).
+
+3. Add a feature gate declaration to `libsyntax/feature_gate.rs` In the active `declare_features` block:
+
+```rust,ignore
+    // description of feature
+    (active, $feature_name, "$current_nightly_version", Some($tracking_issue_number), $edition)
+```
+
+where `$edition` has the type `Option<Edition>`, and is typically just `None`.
+For example:
+
+```rust,ignore
+    // allow '|' at beginning of match arms (RFC 1925)
+(   active, match_beginning_vert, "1.21.0", Some(44101), None),
+```
+
+The current version is not actually important – the important version is when you are stabilizing a feature.
+
+4. Prevent usage of the new feature unless the feature gate is set.
+
+You can check it in most places in the compiler using the expression `tcx.features().$feature_name` (or `sess.features_untracked().borrow().$feature_name` if the tcx is unavailable)
+
+If the feature gate is not set, you should either maintain the pre-feature behavior or raise an error, depending on what makes sense.
+
+5. Add a test to ensure the feature cannot be used without a feature gate, by creating `feature-gate-$feature_name.rs` and `feature-gate-$feature_name.stderr` files under the `src/test/ui/feature-gates` directory.
+
+6. Add a section to the unstable book, in `src/doc/unstable-book/src/language-features/$feature_name.md`.
+
+7. Write a lots of tests for the new feature. PRs without tests will not be accepted!
+
+8. Get your PR reviewed and land it. You have now successfully implemented a feature in Rust!
+
+[the rfcbot guide]: https://github.com/anp/rfcbot-rs/blob/master/CONDUCT.md
+[CONTRIBUTING.md]: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md
+[rustc bug-fix procedure]:
+https://github.com/rust-lang/rust-forge/blob/master/rustc-bug-fix-procedure.md
+[value the stability of Rust]: https://github.com/rust-lang/rfcs/blob/master/text/1122-language-semver.md
+[stability in code]:
```

Comment:
The link to [stability in code]() in [stability section](https://github.com/rust-lang/rust-forge/blob/master/feature-guide.md#stability) refers same document. How to refer the same document?

Reply:
You can add an anchor at line 90 (the beginning of the `stability in code` section):

```html
<a name="stability-in-code"></a>
```

Then, you can link to it with `[stability in code]: #stability-in-code`.